public/app/features/admin/AdminFeatureTogglesAPI.ts

```diff
@@ -26,37 +60,86 @@ const backendSrvBaseQuery =
     }
   };
 
-export const togglesApi = createApi({
-  reducerPath: 'togglesApi',
-  baseQuery: backendSrvBaseQuery({ baseUrl: '/api' }),
-  endpoints: (builder) => ({
-    getManagerState: builder.query<FeatureMgmtState, void>({
-      query: () => ({ url: '/featuremgmt/state' }),
-    }),
-    getFeatureToggles: builder.query<FeatureToggle[], void>({
-      query: () => ({ url: '/featuremgmt' }),
-    }),
-    updateFeatureToggles: builder.mutation<void, FeatureToggle[]>({
-      query: (updatedToggles) => ({
-        url: '/featuremgmt',
-        method: 'POST',
-        body: { featureToggles: updatedToggles },
+const getK8sAPI = () => {
+  return createApi({
+    reducerPath: 'togglesApi',
+    baseQuery: backendSrvBaseQuery({ baseUrl: `/apis/${featureApiVersion}/` }),
+    endpoints: (builder) => ({
+      getManagerState: builder.query<FeatureMgmtState, void>({
+        query: () => ({ url: 'state' }),
+      }),
+      getFeatureToggles: builder.query<FeatureToggle[], void>({
+        query: () => ({ url: 'current' }),
+      }),
+      updateFeatureToggles: builder.mutation<void, FeatureToggle[]>({
+        query: (updatedToggles) => ({
+          url: 'current',
+          method: 'PATCH',
+          body: convertModifiedTogglesToK8sObject(updatedToggles),
+        }),
       }),
     }),
-  }),
-});
-
-type FeatureToggle = {
-  name: string;
-  description?: string;
-  enabled: boolean;
-  readOnly?: boolean;
+  });
 };
 
-type FeatureMgmtState = {
-  restartRequired: boolean;
-  allowEditing: boolean;
+const getLegacyAPI = () => {
+  return createApi({
+    reducerPath: 'togglesApi',
+    baseQuery: backendSrvBaseQuery({ baseUrl: '/api' }),
+    endpoints: (builder) => ({
+      getManagerState: builder.query<FeatureMgmtState, void>({
+        query: () => ({ url: '/featuremgmt/state' }),
+      }),
+      getFeatureToggles: builder.query<FeatureToggle[], void>({
+        query: () => ({ url: '/featuremgmt' }),
+      }),
+      updateFeatureToggles: builder.mutation<void, FeatureToggle[]>({
+        query: (updatedToggles) => ({
+          url: '/featuremgmt',
+          method: 'POST',
+          body: { featureToggles: updatedToggles },
+        }),
+      }),
+    }),
+  });
 };
 
-export const { useGetManagerStateQuery, useGetFeatureTogglesQuery, useUpdateFeatureTogglesMutation } = togglesApi;
+function convertModifiedTogglesToK8sObject(toggles: FeatureToggle[]): ResolvedToggleState {
+  const patchBody: ResolvedToggleState = {
+    kind: 'ResolvedToggleState',
+    enabled: {},
+  };
+  toggles.forEach((t) => {
+    patchBody.enabled[t.name] = t.enabled;
+  });
+  return patchBody;
+}
+
+const togglesApi = config.featureToggles.kubernetesFeatureToggles ? getK8sAPI() : getLegacyAPI();
+let { useGetManagerStateQuery, useGetFeatureTogglesQuery, useUpdateFeatureTogglesMutation } = togglesApi;
+
+if (config.featureToggles.kubernetesFeatureToggles) {
```

Comment:
Using this function wrapping approach, it's transparent to any code calling these functions. However, not sure it is worth the trouble. We could just not export `useGetFeatureTogglesQuery` at all, and instead export an explicit wrapper function.